soc/silabs/silabs_s2/Kconfig

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2025 Silicon Laboratories Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_FAMILY_SILABS_S2
+if MBEDTLS
+
+config PSA_CRYPTO_SW_ONLY
+	bool "PSA Crypto software implementation only"
+
+config PSA_CRYPTO_DRIVER_SILABS_HSE
+	bool "PSA Crypto Driver for Secure Engine"
+	default y if (SOC_SERIES_EFR32MG21 || SOC_SERIES_EFR32MG24 || SOC_SERIES_EFR32BG29 || \
+		      SOC_SERIES_EFR32MG29) && !PSA_CRYPTO_SW_ONLY
```

Comment:
Maybe this is just following some established practice/pattern, but I feel like it'd be more scalable to have each SoC series individually say they want HSE (by selecting some common Kconfig option or perhaps through some DTS property), and then you have a dependency on that single option here instead of having to always update this dependency list.

Reply:
Did it with a DTS property.